### PR TITLE
[DEV-2485] Opensea - list for sale

### PR DESCRIPTION
**Pull request by mihaiberq**
Ready for review. More details in the monorepo PR.

### CODE REVIEW

File: src/nft/ERC721NFT.ts [hunk 1]
```diff
@@ -333,50 +334,37 @@ export class ERC721NFT extends AbstractEvmNFT {
...
+      if (
+        this.item.chainId === Chain.Ethereum ||
+        this.item.chainId === Chain.EthereumGoerli
+      ) {
+        const platformFactory = new PlatformFactory(this.refinable);
+
+        for (const platform of platforms) {
+          const platformInstance = platformFactory.createPlatform(platform);
+
+          const now = Math.floor(Date.now() / 1000);
```
Comment (viet10101988) [inline src/nft/ERC721NFT.ts:346]: this will also conflict with my PR since I refactored this part

Comment (mihaiberq) [reply src/nft/ERC721NFT.ts:346]: yes, yes, whoever gets approved last will have the joy of merging stuff :D :D :D

File: src/refinable/Refinable.ts [hunk 2]
```diff
@@ -128,6 +135,8 @@ export class Refinable {
   }
 
   async init() {
+    const { chainId } = await this.evm.provider.getNetwork();
```
Comment (sneljo1) [inline src/refinable/Refinable.ts:138]: During init this will error because a user might not be authenticated yet

**Review by sneljo1 — CHANGES_REQUESTED** [review]

File: src/platform/OpenseaPlatform.ts [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+      {}
+    );
+  }
+
+  getApprovalAddress(): string {
+    return Seaport.Addresses.Exchange[this.chainId];
+  }
+
+  async buy(offer: PartialOffer, contractAddress: string, tokenId: string) {
+    throw new Error("Buy method not implemented yet");
```
Comment (sneljo1) [inline src/platform/OpenseaPlatform.ts:143]: Is it or is it not implemented?

**Review by sneljo1 — APPROVED** [review]

File: src/platform/OpenseaPlatform.ts (hunk 3 above)
Comment (mihaiberq) [reply src/platform/OpenseaPlatform.ts:143]: initial copy-paste from putForSale, params still have to be changed to match the new ones. but the gist of it should be there

**Pull request merged**